libyul/optimiser/DataFlowAnalyzer.cpp

```diff
@@ -346,17 +347,19 @@ void DataFlowAnalyzer::clearValues(std::set<YulName> _variables)
 	});
 
 	// Also clear variables that reference variables to be cleared.
-	std::set<YulName> referencingVariables;
-	for (auto const& variableToClear: _variables)
-		for (auto const& [ref, names]: m_state.references)
-			if (names.count(variableToClear))
-				referencingVariables.emplace(ref);
+	std::set<YulName> referencingVariablesToClear;
+	std::vector const sortedVariablesToClear(_variables.begin(), _variables.end());
+	for (auto const& [ref, names]: m_state.sortedReferences)
```

Comment:
Can you also rename `ref`, `names` and `_variables` as suggested in https://github.com/ethereum/solidity/pull/15842#discussion_r1954981807? They were not added by the PR, but we come back to this place surprisingly often (it's 3th or 4th PR touching it I remember over the years) so it's worth keeping it readable :)

Reply:
Oh yeah, somehow I completely missed that. Fixed!